packages/autoupdate/client_versions.js

```diff
@@ -0,0 +1,57 @@
+export class ClientVersions {
+  constructor() {
+    this.versions = new Map();
+    this.watchers = new Set();
+
+    // A Livedata store for use with `Meteor.connection.registerStore`.
+    this.store = {
+      update: ({ id, msg, fields }) => {
+        if (msg === "added" || msg === "changed") {
+          this.set(id, fields);
+        }
+      }
+    };
+  }
+
+  get(id) {
+    return this.versions.get(id);
+  }
+
+  // Adds or updates a version document and invokes all registered callbacks
+  // with the added/updated document.
+  set(id, fields) {
+    let version = this.versions.get(id);
+    let isNew = false;
+
+    if (version) {
+      Object.assign(version, fields);
+    } else {
+      version = {
+        _id: id,
+        ...fields
+      };
+
+      isNew = true;
+      this.versions.set(id, version);
+    }
+
+    this.watchers.forEach((callback) => {
+      callback(version, isNew);
+    });
+  }
+
+  // Registers a callback that will be invoked when a version document is added
+  // or changed. Call the function returned by `watch` to remove the callback.
+  // If `skipInitial` is false, the callback will be invoked immediately for all
+  // existing version documents.
+  watch(callback, skipInitial = false) {
+    if (!skipInitial) {
+      this.versions.forEach((version) => {
+        callback(version, true);
+      });
+    }
```

Comment:
I think we should be careful to delay calling the callback until the next tick of the event loop, so that the `stop` callback is returned (and can be used to initialize a `let` or `const` variable) before the callback fires. See https://github.com/meteor/meteor/commit/f2c864ec667fc5f85e5ecb2e6379262e0dff093b for a similar fix.